src/edu/csus/ecs/pc2/core/scoring/DefaultScoringAlgorithm.java

```diff
@@ -579,15 +581,34 @@ private void createStandingXML (TreeMap<StandingsRecord, StandingsRecord> treeMa
         Hashtable<Group, Integer> groupIndexHash = new Hashtable<Group, Integer>();
         int groupCount = 0;
         for (Group group : groups) {
-            // no refence to groups that should not be displayed on scoreboard
+            // no reference to groups that should not be displayed on scoreboard
             if (!group.isDisplayOnScoreboard()) {
                 continue;
             }
             groupHash.put(group.getElementId(), group);
             groupIndexHash.put(group, Integer.valueOf(groupCount));
             groupCount++;
         }
-        
+        Hashtable<Group, Integer> divisionIndexHash = new Hashtable<Group, Integer>();
+        int divisionCount = 0;
+        // TODO this bit should in the future can probably go away when divisions are supported better.
+        int lastFound = 0;
+        for (Group group : groups) {
+            // no reference to groups that should not be displayed on scoreboard
+            if (!group.isDisplayOnScoreboard()) {
+                continue;
+            }
+            Pattern pattern = Pattern.compile("D\\d+$");
+            Matcher matcher = pattern.matcher(group.getDisplayName());
+            if (matcher.find()) {
+                Integer found = Integer.parseInt(matcher.group(0).substring(1));
+                if (found.intValue() > lastFound) {
+                    lastFound = found.intValue();
+                }
```

Comment:
//insert the current Group into the hashtable with its Division number as the value